### PR TITLE
Fix Mongo SQL type conversion asymmetry

### PR DESCRIPTION
**Pull request by TomaszGaweda**
Fixes [HZ-2098](https://hazelcast.atlassian.net/browse/HZ-2098)

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible
- [N/A] Send backports/forwardports if fix needs to be applied to past/future releases
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc

[HZ-2098]: https://hazelcast.atlassian.net/browse/HZ-2098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TomaszGaweda** [comment]
run-lab-run

**Review by ivanthescientist — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/ConversionsToBson.java [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+            return new ObjectId(v.toBigInteger().toString(HEX_RADIX));
+        }
+        if (value instanceof BigInteger) {
+            BigInteger v = (BigInteger) value;
+            return new ObjectId(v.toString(HEX_RADIX));
+        }
+        return null;
+    }
+
+    private static Object convertToArray(Object value) {
```
Comment (orcunc) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/ConversionsToBson.java:217]: Return type in the signature could be collection

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/ConversionsToBson.java [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+        if (value instanceof Collection) {
+            return value;
+        }
+
+        return singletonList(value);
+    }
+
+    private static Object convertToBoolean(Object value) {
```
Comment (orcunc) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/ConversionsToBson.java:225]: Return type in the signature could be boolean

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/ConversionsToBson.java [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+        if (value instanceof BsonMaxKey) {
+            return new MaxKey();
+        }
+        if (value instanceof String && "MaxKey".equalsIgnoreCase((String) value)) {
+            return new MaxKey();
+        }
+        return null;
+    }
+//
+//    private static Object localDateTimeToBson(LocalDateTime value, BsonType bsonType) {
```
Comment (orcunc) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/ConversionsToBson.java:407]: Commented out code necessary ?

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoStreamSqlConnectorTest.java
Comment (orcunc) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoStreamSqlConnectorTest.java:1]: Close the MongoClient in readsFromMongo() maybe ?

**Review by orcunc — APPROVED** [review]

**Pull request merged**